types/pair/README.md

```diff
@@ -0,0 +1,15 @@
+# `std::pair`
+
+## Fields
+
+ * `Int32_String`: `std::pair<std::int32_t, std::string>`
+ * `Variant_Vector`: `std::pair<std::variant<std::int32_t, std::string>, std::vector<std::int32_t>>`
+ * `Pair`: `std::pair<std::pair<std::int32_t, std::string>, std::int32_t>`
+ * `VectorPair`: `std::vector<std::pair<std::int32_t, std::string>>`
```

Comment:
One more combination might (or might not) be interesting:
```std::pair<std::vector<std::pair<std::variant<std::int32_t, std::string>, std::int32_t>,std::int32_t>```

Reply:
I'm not sure I see what this adds:
1. `std::pair<std::vector` we test in `Variant_Vector` (as the second element type)
2. `std::vector<std::pair` we test here in `VectorPair` (was missing initially)
3. `std::pair<std::variant` we test in `Variant_Vector` (as the first element type)

Reply:
For the binary format, which we are aiming to test, there is no dependency between the surrounding collection and the inner variant. `std::vector<std::variant` will be added as part of #48 (inside `types/variant`).

Reply:
Isn't the header which contains the list of field/columns and directly (or indirectly) their relationship part of the binary format?

Reply:
Yes to some extent, that's why we are testing *one* level of nesting. But from the binary format, there should be no dependency that crosses two field chains. That's the assumption that we make, that we discussed many times by now.

Reply:
[The variant was an auxiliary part and indeed is fine to be checked in #48, the point of  `https://github.com/root-project/rntuple-validation/pull/46/files#r2078264823` was to probe more than one level of nesting.]

Reply:
I still don't see which part of the binary format depends on two levels of (record) nesting. Can you explain why you want to see this test added? Which case does it exercise that is not present individually?

Reply:
How is  the fields/columns hierarchy/nesting information stored in the headers?